packages/web/src/utils/validators.ts

```diff
@@ -0,0 +1,256 @@
+          // Randomizes the order of validators on every load
+          order: Math.random(),
+          pinned: isPinned(account.address),
+          name: account.name,
+          address: account.address,
+          usd: weiToDecimal(+account.usd),
+          celo: weiToDecimal(+account.lockedGold),
+          receivableRaw: weiToDecimal(+receivableVotes),
+          receivableVotes: receivableVotesPer.toString(),
+          votesRaw: weiToDecimal(+votes),
+          votes: votesPer.toString(),
+          votesAbsolute: votesAbsolutePer.toString(),
+          commission: (+commission * 100) / 10 ** 24,
+          rewards,
+          rewardsStyle,
+          numMembers,
+          claims: getClaims(account.claims),
+          validators: affiliates.edges.map(({ node: validator }) => {
+            const {
+              address,
+              lastElected,
+              lastOnline,
+              name,
+              usd,
+              lockedGold,
+              score,
+              attestationsFulfilled,
+              attestationsRequested,
+            } = validator
+            return {
+              name,
+              address,
+              usd: weiToDecimal(+usd),
+              celo: weiToDecimal(+lockedGold),
+              elected: lastElected >= latestBlock,
+              online: lastOnline >= latestBlock,
+              uptime: (+score * 100) / 10 ** 24,
+              attestation: Math.max(0, attestationsFulfilled / (attestationsRequested || -1)) * 100,
+              neverElected: !lastElected && !attestationsRequested,
+              claims: getClaims(validator.account.claims),
+            }
+          }),
+        }
+      }
+    )
+    .map((group, id) => {
+      const data = group.validators.reduce(
+        ({ elected, online, total, uptime }, validator) => ({
+          elected: elected + +validator.elected,
+          online: online + +validator.online,
+          total: total + 1,
+          uptime: uptime + validator.uptime,
+        }),
+        { elected: 0, online: 0, total: 0, uptime: 0 }
+      )
+      data.uptime = data.uptime / group.validators.length
+      return {
+        id,
+        ...group,
+        ...data,
+      }
+    })
+}
+
+/**
+ * Sorts validator data by specified key, with pinned validators at the beginning
+ * @param {ValidatorsListProps['data']} data The validators data
+ * @param {boolean}       asc Whether the data should be sorted in ascending or descending order
+ * @param {string}        key The key to sort by (e.g. name, votes, lockedCelo, etc)
+ * @return {CeloGroup}    The group of validators with formatted data
+ */
+export function sortData(data: CeloGroup[], asc: boolean, key: string) {
+  const accessor = orderAccessors[key]
+  const dir = asc ? 1 : -1
+
+  const compare = (a, b): number => {
+    if (a === null) {
+      return 1
+    }
+    if (b === null) {
+      return -1
+    }
+    return a > b ? 1 : -1
```

Comment:
thoughts on returning 0 for cases where they are equal?

Reply:
Hm, I don't think it's necessary.  Thought maybe just returning `a - b` would be a good way to simplify, but that doesn't work for some reason

Reply:
for `a - b` to work that operation would need to return a number but `a` and `b` could be strings.

Reply:
That would work for numbers.  It sorts it based on if it's negative, positive or 0.  But you're right, it doesn't work for strings.  I think this is fine as is.